target/linux/realtek/files-5.15/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -1922,7 +1923,7 @@ static int rtl930x_mdio_reset(struct mii_bus *bus)
 	for (int i = 0; i < RTL930X_CPU_PORT; i++) {
 		int pos;
 
-		if (priv->smi_bus[i] > 3)
+		if (priv->smi_bus[i] > MAX_SMI_BUSSES)
```

Comment:
This also changes the check from "> 3> to "> 4".

Reply:
Well spotted

```suggestion
		if (priv->smi_bus[i] >= MAX_SMI_BUSSES)
```